app/controllers/application_controller.rb

```diff
@@ -167,7 +167,7 @@ def respond_with_request_for_authentication
 
   def redirect_permanently_to(url = nil, **args)
     if url
-      redirect_to(url + internal_nav_param, status: :moved_permanently)
+      redirect_to(URL.url(url) + internal_nav_param, status: :moved_permanently)
```

Comment:
We've had a lot of problems with flaky tests lately, so apologies for not catching this sooner, but the tests aren't happy because this changes many "relative (path) redirects" to being "absolute (host+path) redirects" in a way that the tests aren't presently configured to support, because Rails test environment defaults assume the app's hostname is `www.example.com` but our test environment is (partially) setup such that the test environment assumes the hostname is `forem.test`.

I actually think "absolute (host+path) redirect" is technically a better implementation for `redirect_permanently` but we'd need to update (at least) all of the failing tests to correctly set the domain's request to `forem.test`. (Ideally, we would have already set that up in `rails_helper`, but it seems we have not.)

Reply:
Hi, in that case, does directly using `Addressable::URI` work? Namely:

``` ruby
Addressable::URI.parse(url + internal_nav_param).normalize.to_s
```

(Sorry for the force-push, I pushed an irrelevant commit and GitHub mirrored all branches.)